### ISSUE TITLE
azure-search-documents Readme and samples issues

### ISSUE THREAD

**Issue by v-xuto** [Azure/azure-sdk-for-python#12506]
1.
Section [Link1](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/search/azure-search-documents#querying) and [Link2](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/search/azure-search-documents#async-apis):
![image](https://user-images.githubusercontent.com/57166602/87274840-4d488100-c50f-11ea-9df2-7deeb68512e2.png)

Reason:
KeyError: 'hotelId', 'hotelName'.

Suggestion:
```
print("{}: {})".format(result["HotelId"], result["HotelName"]))
```
2.
Section [link](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/search/azure-search-documents#adding-documents-to-your-index):
![image](https://user-images.githubusercontent.com/57166602/87274894-78cb6b80-c50f-11ea-80bd-5d435d6e3808.png)

Reason:
NameError: name 'client' is not defined.

Suggestion:
```
search_client = SearchClient(endpoint, index_name, credential=AzureKeyCredential(key))
result = search_client.upload_documents(documents=[DOCUMENT])
```
3.
Section [Link](https://github.com/Azure/azure-sdk-for-python/tree/master/sdk/search/azure-search-documents#authenticate-the-client):
![image](https://user-images.githubusercontent.com/57166602/87274986-b6c88f80-c50f-11ea-84f7-15ae2501a030.png)

Suggestion:
Remove extra ';' at the end of the sentence.
4.
Section [Link](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/search/azure-search-documents/samples/sample_indexer_datasource_skillset.py#L78):
![image](https://user-images.githubusercontent.com/57166602/87276717-7bc85b00-c513-11ea-842a-b9fd95a61d26.png)

Reason:
AttributeError: 'SearchIndexerClient' object has no attribute 'create_datasource'.

Suggestion:
```
def _create_datasource():
    # Here we create a datasource. As mentioned in the description we have stored it in
    # "searchcontainer"
    ds_client = SearchIndexerClient(service_endpoint, AzureKeyCredential(key))
    container = SearchIndexerDataContainer(name='searchcontainer')
    data_source_connection = SearchIndexerDataSourceConnection(
        name="hotel-datasource",
        type="azureblob",
        connection_string=connection_string,
        container=container
    )
    result = ds_client.create_data_source_connection(data_source_connection)
    return result
```
5.
Section [Link](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/search/azure-search-documents/samples/async_samples/sample_indexers_operations_async.py#L41):
![image](https://user-images.githubusercontent.com/57166602/87276817-af0aea00-c513-11ea-8493-622930628c64.png)

Suggestion1:
Update `SearchIndexerClientt ` to `SearchIndexClient`.

Suggestion2:
Update `ind_client ` to `indexers_client`.

Suggestion3:
Update `indexer-hotels` to `hotels`.

**Comment by v-xuto** [comment]
@jongio for notification.

**Comment by kaerm** [comment]
@v-xuto thanks for reviewing this, @xiangyan99 can you take a look at this

**Comment by xiangyan99** [comment]
Fixed in #12595
